File: android/src/main/kotlin/dev/jonathanvegasp/result_channel/Reader.kt
package dev.jonathanvegasp.result_channel

interface Reader {
    fun byte(): Byte
    fun char(): Char
    fun int(): Int
    fun long(): Long
    fun float(): Float
    fun double(): Double
    fun string(size: Int): String
    fun byteArray(size: Int): ByteArray
    fun intArray(size: Int): IntArray
    fun longArray(size: Int): LongArray
    fun floatArray(size: Int): FloatArray
    fun doubleArray(size: Int): DoubleArray
}

File: ios/Classes/Reader.swift
import Foundation

public protocol Reader {
    func byte() -> UInt8
    func char() -> UInt16
    func int() -> Int32
    func long() -> Int64
    func float() -> Float
    func double() -> Double
    func string(size: Int) -> String
    func byteArray(size: Int) -> Data
    func intArray(size: Int) -> [Int32]
    func longArray(size: Int) -> [Int64]
    func floatArray(size: Int) -> [Float]
    func doubleArray(size: Int) -> [Double]
}
